pydantic_ai_slim/pydantic_ai/models/bedrock.py

```diff
@@ -478,17 +482,58 @@ def _map_inference_config(
 
         return inference_config
 
-    def _map_tool_config(self, model_request_parameters: ModelRequestParameters) -> ToolConfigurationTypeDef | None:
-        tools = self._get_tools(model_request_parameters)
-        if not tools:
+    def _map_tool_config(
+        self,
+        model_request_parameters: ModelRequestParameters,
+        model_settings: BedrockModelSettings | None,
+    ) -> ToolConfigurationTypeDef | None:
+        resolved = _resolve_tool_choice(model_settings, model_request_parameters)
+        function_tools = model_request_parameters.function_tools
+        output_tools = model_request_parameters.output_tools
+
+        if resolved is None:
+            tool_defs_to_send = [*function_tools, *output_tools]
+        else:
+            tool_defs_to_send = resolved.filter_tools(function_tools, output_tools)
+
+        if not tool_defs_to_send:
             return None
 
+        tools = [self._map_tool_definition(t) for t in tool_defs_to_send]
         tool_choice: ToolChoiceTypeDef
-        if not model_request_parameters.allow_text_output:
+
+        if resolved is None:
+            # Default behavior: infer from allow_text_output
+            if not model_request_parameters.allow_text_output:
+                tool_choice = {'any': {}}
+            else:
+                tool_choice = {'auto': {}}
+
+        elif resolved.mode == 'auto':
+            if not model_request_parameters.allow_text_output:
+                tool_choice = {'any': {}}
+            else:
+                tool_choice = {'auto': {}}
+
+        elif resolved.mode == 'required':
             tool_choice = {'any': {}}
-        else:
+
+        elif resolved.mode == 'none':
+            # We've already filtered to only output tools, use 'auto' to let model choose
             tool_choice = {'auto': {}}
```

Comment:
If `not model_request_parameters.allow_text_output:`, shouldn't we send `tool_choice = {'any': {}}` as above, so we force an output tool to be used? `'auto'` would allow text output which we don't want